File: 0344-reverse-string/0344-reverse-string.cpp
#define n s.size()

static const auto fast_io = [](){
    ios:: sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);
    return 0;
} ();


class Solution {
public:
    void reverseString(vector<char>& s) {
        int l =0;
        int r = n-1;

        while(l<r) {
            swap (s[l], s[r]);
            l++;
            r--;
        }
    }
};

File: 0344-reverse-string/0344-reverse-string.py
class Solution:
    # 4. Reverse String (LeetCode 344)
    def reverseString(self, s: list[str]) -> None:
        # Initialize two pointers at opposite ends of the list
        left, right = 0, len(s) - 1
        
        # Swap characters moving towards the center
        while left < right:
            s[left], s[right] = s[right], s[left]
            left += 1
            right -= 1